core/src/main/java/org/elasticsearch/node/service/NodeService.java

```diff
@@ -81,12 +86,14 @@ public NodeService(Settings settings, ThreadPool threadPool, MonitorService moni
         this.version = version;
         this.pluginService = pluginService;
         this.circuitBreakerService = circuitBreakerService;
+        this.ingestService = new IngestService(settings, threadPool, clusterService, processorsRegistry);
```

Comment:
can we register the PipelineStore here as a listener I think this is the right separation